ietfdata/datatracker.py

```diff
@@ -400,6 +400,54 @@ class DocumentEvent:
     type            : str
 
 
+@dataclass(frozen=True)
+class RelationshipTypeURI(URI):
+    def __post_init__(self) -> None:
+        assert self.uri.startswith("/api/v1/name/docrelationshipname/")
+
+
+@dataclass(frozen=True)
+class RelationshipType:
+    resource_uri   : RelationshipTypeURI
+    slug           : str
+    desc           : str
+    name           : str
+    used           : bool
+    order          : int
+    revname        : str
+
+
+@dataclass(frozen=True)
+class RelatedDocumentURI(URI):
+    def __post_init__(self) -> None:
+        assert self.uri.startswith("/api/v1/doc/relateddocument/")
+
+
+@dataclass(frozen=True)
+class RelatedDocument:
+    id              : int
+    relationship    : RelationshipTypeURI
+    resource_uri    : RelatedDocumentURI
+    source          : DocumentURI
+    target          : DocumentAliasURI
+
+    def related_document_source(self, 
+            source               : Optional[Document]         = None, 
+            relationship_type    : Optional[RelationshipType] = None) -> str:
+        url = "?source=" + source
+        if relationship_type is not None:
+            url = url + "&relationship=" + relationship_type
+        return url
+
+    def related_document_target(self, 
+            target               : Optional[Document]         = None, 
+            relationship_type    : Optional[RelationshipType] = None) -> str:
+        url = "?target=" + target
+        if relationship_type is not None:
+            url = url + "&relationship=" + relationship_type
+        return url
+
+
 # ---------------------------------------------------------------------------------------------------------------------------------
```

Comment:
It's not clear why you've introduced these functions.

Reply:
Removed functions